services/base.js

```diff
@@ -11,6 +11,7 @@ const {
   InvalidParameter,
   Deprecated,
 } = require('./errors')
+const coalesce = require('../lib/coalesce')
```

Comment:
(I ran out of time, but I'll add a few comments to this file and the ones below when I can get back to it.)